toxcore/events/dht_get_nodes_response.c

```diff
@@ -0,0 +1,237 @@
+/* SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright © 2022 The TokTok team.
+ */
+
+#include "events_alloc.h"
+
+#include <assert.h>
+#include <string.h>
+
+#include "../bin_pack.h"
+#include "../bin_unpack.h"
+#include "../ccompat.h"
+#include "../mem.h"
+#include "../tox.h"
+#include "../tox_events.h"
+#include "../tox_private.h"
+
+
+/*****************************************************
+ *
+ * :: struct and accessors
+ *
+ *****************************************************/
+
+
+struct Tox_Event_Dht_Get_Nodes_Response {
+    uint8_t public_key[TOX_PUBLIC_KEY_SIZE];
+    uint8_t *ip;
+    uint32_t ip_length;
+    uint16_t port;
+};
+
+non_null()
+static bool tox_event_dht_get_nodes_response_set_public_key(Tox_Event_Dht_Get_Nodes_Response *dht_get_nodes_response, const uint8_t *public_key)
+{
+    assert(dht_get_nodes_response != nullptr);
```

Comment:
Are the asserts necessary even with the `non_null`-annotations? If so, should all pointer arguments be asserted to be non-null?

Reply:
No, they are leftovers from when I first implemented this. I've removed them now.